src/social_wallet_api/handler.clj

```diff
@@ -184,126 +195,146 @@
                                                       :currency (:connection query)))))))
              blockchain-transactions)))
 
+(defn wrap-auth [handler]
+  (fn [request]
+    (if @client
+      (if (= (or (-> request :headers (get "X-API-Key"))
+                 (-> request :headers (get "x-api-key")))
```

Comment:
according to RFC 2616 - "Hypertext Transfer Protocol -- HTTP/1.1" headers are case insensitive and it seems that by default in ring they are lowercase. I presume you detected a different behavior here? however my advice is to put the lowercase get first since it may well be the most frequent.

Reply:
You are totally right and also proved by [this test](https://github.com/Commonfare-net/social-wallet-api/blob/feature/api-key-rebased/test/social_wallet_api/test/api_key.clj#L78) so just removed upper case test